Dockerfile

```diff
@@ -1,23 +1,42 @@
-FROM node:20-slim
-
+FROM node:20-slim as base
 RUN apt-get update && \
     apt-get install -y python3 curl git build-essential psmisc && \
     apt-get clean
 
 COPY . /app
 WORKDIR /app
 
-ENV DEV_CHAIN_ID=313371
-ENV PNPM_HOME="/pnpm"
-ENV PATH="$PNPM_HOME:$PATH"
-RUN corepack enable
 
 # Check when foundry supports dumping events
 # https://github.com/foundry-rs/foundry/issues/5906
 RUN curl -L https://foundry.paradigm.xyz | bash
-RUN ~/.foundry/bin/foundryup
 
+# install a previous versions that doesn't have the rpc method hardhat_metadata.
+# hardhat expects an integer but the output from anvil is an hex string
+RUN ~/.foundry/bin/foundryup -v nightly-34f684ddfacc5b2ed371353ba6f730c485616ffe
```

Comment:
Do you remember what problem arose that required this and how it manifested? Might be a good time to see if it can be addressed, so as to be able to track foundry's latest.

Reply:
Nevermind, I think I reproduced it:

```
dev-setup-2-1   | TypeError: Cannot read properties of null (reading 'chainId')
dev-setup-2-1   |     at new Manifest (/app/node_modules/.pnpm/@openzeppelin+upgrades-core@1.24.1/node_modules/@openzeppelin/upgrades-core/src/manifest.ts:133:65)
dev-setup-2-1   |     at Function.forNetwork (/app/node_modules/.pnpm/@openzeppelin+upgrades-core@1.24.1/node_modules/@openzeppelin/upgrades-core/src/manifest.ts:104:14)
dev-setup-2-1   |     at processTicksAndRejections (node:internal/process/task_queues:95:5)
dev-setup-2-1   |     at async Proxy.deployProxy (/app/node_modules/.pnpm/@openzeppelin+hardhat-upgrades@1.22.1_@nomiclabs+hardhat-ethers@2.2.2_@nomiclabs+hardhat-ethe_vpojyqhxd3twqv4stmecthya54/node_modules/@openzeppelin/hardhat-upgrades/src/deploy-proxy.ts:33:22)
dev-setup-2-1   |     at async main (/app/scripts/projectRegistry/deployProjectRegistry.ts:36:20)
```